daos-stack/daos
# DAOS-11258 test: Recording launch.py output in logs (#10082)

Modify method check_rank_state in ftest/util/server_util.py
to support multiple ranks

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

Updated test code to run soak via a Jenkins build in CI
Updates include:
- Allow ftest/main to include up to 24 nodes
- Update post provisioning to easily add *-testing branch
- Create DAOS_APP_DIR env within the testing framework
- Changed vpic default mpi to mpich
- Fixed an issue in soak cleanup where it was removing the shared tmp dir

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

Adding logging debug messages for launch.py outut in the Jenkins test
results and including all test post-processing in a separate launch.log
stored with the test results.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

Disable the bandit check for random.choice() usage in the
harness/advance.py test_launch_failures test.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

Increasing the test logs threshold size check to match the log rollover
size: 1GB -> 2GB.  Increasing the timeout for the cart_logtest command
and removing any empty log files prior to running the cart_logtest
command. Also resolving DAOS-11812, where non-contiguous node sets where
not processed correctly when updating test yaml files.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

If launch.py experienced an error before it could establish a file
handler for logging debug messages to a file, it could fail without
providing the reason for the failure.  This could happen when running
manually if the python avocado libraries where not installed or
misconfigured. The console logger will now report all messages until the
file handler is configured, at which point the console logger will be
updated to only report info messages.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

Test-tag: pr daily_regression soak_smoke criticalintegration test_core_files test_launch_failures

### Before requesting gatekeeper:


### Gatekeeper:

* [ ] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
